### ISSUE TITLE
Making AdHoc ES Requests

### ISSUE THREAD

**Issue by MattSurabian** [searchkit/searchkit#513]
While trying to work out the best way to get autocomplete/typeahead working on refinement lists and the searchbox, I found [the WIP repo in the searchkit org for autocomplete](https://github.com/searchkit/searchkit-autocomplete), read a few issues, and checked out the core code.

I ended up trying a few different things, like building a custom accessor and making a parent component that shared state between a refinement list and react-typeahead component, etc. Ultimately what I realized was that in the particular application I'm working on the cleanest route forward was to make an adhoc request to elastic from an extended searchkit component that was able to render the response in standard search suggestion style. That adhoc request would need information about any existing application state to ensure the typeahead suggestions were within the current result.

I ended up using axios for the request since it was already in my bundle as a dep of searchkit and building the query to fetch a list of autocomplete options with an ES6 template string that borrowed information from the searchkitmanager's query object and properties passed in through reach, like what fields to suggest on, etc.

_Is there a better way to do this than using axios directly and working with template strings?_

I read some of the core code where the transport object and request are built up but it seemed like there wasn't a clear path to fire a single request to ES without the response altering the searchkitmanager's state.

**Comment by joemcelroy** [comment]
Yeh for now thats the best approach by using axios directly. It is possible to also create a new searchkit Instance within the accessor that is independent from the main sk instance. That was the route I was trying to go but it probably not an easy route for those who dont know the code. I can try and help if you do follow that route tho.

**Comment by MattSurabian** [comment]
Thanks @joemcelroy! I did play around a bit with creating a new searchkit instance but ultimately it was adding a lot of complexity for the folks that need to maintain this application. I think a core searchkit component doing that makes a lot of sense, but am planning to stick with using axios directly.

Another thing I was considering was trying to use `call` on some of the existing searchkit prototype methods to make working with the returned results a little easier but that's something I've got to play around with more.

Either way thanks for the quick response :+1: and great component library.
